crates/aptos/src/op/key.rs

```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
             KeyType::Ed25519 => self.save_params.save_key(&ed25519_key, "ed22519"),
         }
     }
+
+    /// A test friendly typed key generation for x25519 keys.
+    pub fn generate_x25519(
+        encoding_type: EncodingType,
+        key_file: &Path,
+    ) -> Result<(x25519::PrivateKey, x25519::PublicKey), Error> {
+        let args = format!(
+            "generate --key-type {key_type:?} --key-file {key_file} --encoding {encoding_type:?} --assume-yes",
+            key_type = KeyType::X25519,
+            key_file = key_file.to_str().unwrap(),
+            encoding_type = encoding_type,
+        );
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
+        ))
+    }
+
+    /// A test friendly typed key generation for e25519 keys.
+    pub fn generate_ed25519(
+        encoding_type: EncodingType,
+        key_file: &Path,
+    ) -> Result<(ed25519::Ed25519PrivateKey, ed25519::Ed25519PublicKey), Error> {
+        let args = format!(
+            "generate --key-type {key_type:?} --key-file {key_file} --encoding {encoding_type:?} --assume-yes",
+            key_type = KeyType::Ed25519,
+            key_file = key_file.to_str().unwrap(),
+            encoding_type = encoding_type,
+        );
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
+        ))
+    }
+
+    /// Generates an `Ed25519PrivateKey` without saving it to disk
+    pub fn generate_ed25519_locally() -> ed25519::Ed25519PrivateKey {
```

Comment:
nit: maybe `new_in_memory_ed25519` or `generate_ed25519_in_mem`? `locally` made me think we're using the local random number generator (e.g., OS) and not that we weren't writing it to a file.